ENGAGEHF/Dashboard/Notifications/NotificationManager.swift

```diff
@@ -0,0 +1,158 @@
+    
+    var notifications: [Notification] = []
+    
+    
+    func configure() {
+        if ProcessInfo.processInfo.isPreviewSimulator {
+            let dummyNotification = Notification(
+                type: "Mock Notification",
+                title: "Weight Recorded",
+                description: "A weight measurement has been recorded.",
+                id: String(describing: UUID())
+            )
+            notifications.append(dummyNotification)
+            return
+        }
+        
+        authStateDidChangeListenerHandle = Auth.auth().addStateDidChangeListener { [weak self] _, user in
+            self?.registerSnapshotListener(user: user)
+        }
+        self.registerSnapshotListener(user: Auth.auth().currentUser)
+    }
+    
+    // Call on initialization
+    //
+    // Creates a snapshot listener to save new notifications to the manager
+    // as they are added to the user's directory in Firebase
+    func registerSnapshotListener(user: User?) {
+        logger.info("Initializing notifiation snapshot listener...")
+        
+        // Remove previous snapshot listener for the user before creating new one
+        snapshotListener?.remove()
+        guard let uid = user?.uid else {
+            return
+        }
+        
+        let db = Firestore.firestore()
+        
+        // Ignore notifications older than expirationDate
+        guard let thresholdDate = Calendar.current.date(byAdding: .day, value: -expirationDate, to: .now) else {
+            logger.error("Unable to get threshold date: \(FetchingError.invalidTimestamp)")
+            return
+        }
+        
+        let thesholdTimeStamp = Timestamp(date: thresholdDate)
+        
+        // Set a snapshot listener on the query for valid notifications
+        db.collection("users").document(uid).collection("notifications")
+            .whereField("created", isGreaterThan: thesholdTimeStamp)
+            .addSnapshotListener { querySnapshot, error in
+                guard let documents = querySnapshot?.documents else {
+                    self.logger.error("Error fetching documents: \(error!)")
+                    return
+                }
+                
+                self.notifications = documents.compactMap {
+                    if $0.get("completed") == nil {
+                        return Notification(
+                            type: String(describing: $0["type"] ?? "Unknown"),
+                            title: String(describing: $0["title"] ?? "Unknown"),
+                            description: String(describing: $0["description"] ?? "Unknown"),
+                            id: $0.documentID
+                        )
+                    }
+                    return nil
+                }
+                
+                self.logger.debug("Notifications updated")
+            }
+    }
+    
+    func add(_ notification: Notification) async {
+        await standard.add(notification: notification)
+    }
```

Comment:
I am wondering where this function is used and how it could be used in the future. If there is no specific use case I would suggest to remove it for now here and in the Standard.